### PR TITLE
fix: lua script limit in /tmp/lua dir logic error

### PR DESCRIPTION
**Pull request by dudebing99**

### CODE REVIEW

**Comment by dudebing99** [comment]
```code
root@ubuntu:~/WaykiChain/wayki/node1# ll /tmp/lua/../../root/a.lua 
-rw-r--r-- 1 root root 315  2月 27 11:28 /tmp/lua/../../root/a.lua
root@ubuntu:~/WaykiChain/wayki/node1# ./node1 -datadir=. registercontracttx 0-1 "/tmp/lua/../../root/a.lua" 10000
error: {"code":-18,"message":"Lua Script file not inside /tmp/lua dir or its subdir!"}

root@ubuntu:~/WaykiChain/wayki/node1# ll /tmp/lua/../../root/x.lua                                                
ls: 无法访问/tmp/lua/../../root/x.lua: 没有那个文件或目录
root@ubuntu:~/WaykiChain/wayki/node1# ./node1 -datadir=. registercontracttx 0-1 "/tmp/lua/../../root/x.lua" 10000 
error: {"code":-19,"message":"Lua Script file not exist!"}
```

File: src/rpc/rpctx.cpp [hunk 1]
```diff
@@ -586,8 +586,11 @@ Value registercontracttx(const Array& params, bool fHelp)
 
     RPCTypeCheck(params, list_of(str_type)(str_type)(int_type)(int_type)(str_type));
 
-    string luaScriptFilePath = params[1].get_str();
-    if (luaScriptFilePath.find("/tmp/lua") == std::string::npos)
+    string luaScriptFilePath = GetAbsolutePath(params[1].get_str()).string();
+    if (luaScriptFilePath.empty())
+        throw JSONRPCError(RPC_SCRIPT_FILEPATH_NOT_EXIST, "Lua Script file not exist!");
+
+    if (luaScriptFilePath.find("/tmp/lua") != 0)
```
Comment (chenxinhua2018) [inline src/rpc/rpctx.cpp]: should use the startswith("/tmp/lua") method
example:
    const std::string TEMP_LUA_DIR = "/tmp/lua";
    if (luaScriptFilePath.compare(0, TEMP_LUA_DIR.size(), TEMP_LUA_DIR) != 0) {

**Comment by dudebing99** [comment]
```code
root@ubuntu:~/WaykiChain/wayki/node1# ls /root/tmp/lua/a.lua 
/root/tmp/lua/a.lua
root@ubuntu:~/WaykiChain/wayki/node1# ./node1 -datadir=. registercontracttx 0-1 "/root/tmp/lua/a.lua" 10000
error: {"code":-18,"message":"Lua Script file not inside /tmp/lua dir or its subdir!"}
```

File: src/rpc/rpctx.cpp [hunk 2]
```diff
@@ -2406,8 +2409,11 @@ Value genregistercontractraw(const Array& params, bool fHelp) {
     vector<unsigned char> vscript;
     int flag = params[2].get_bool();
     if (0 == flag) {
-        string luaScriptFilePath = params[3].get_str();
-	    if (luaScriptFilePath.find("//tmp//lua") != std::string::npos)
+        string luaScriptFilePath = GetAbsolutePath(params[3].get_str()).string();
+        if (luaScriptFilePath.empty())
+            throw JSONRPCError(RPC_SCRIPT_FILEPATH_NOT_EXIST, "Lua Script file not exist!");
+
+        if (luaScriptFilePath.find("/tmp/lua") != 0)
```
Comment (chenxinhua2018) [inline src/rpc/rpctx.cpp]: the same problem as above

**Pull request merged**